mediawiki/mediawikipage.py

```diff
@@ -418,6 +419,49 @@ def sections(self):
                     self._sections.append(obj)
         return self._sections
 
+    def __parse_sections(self, sections):
+        nested = collections.OrderedDict()
+        name_stack = []
+        previous_level = None
+        for section in sections:
+
+            name = section['anchor']
+            level = int(section['level'])
+
+            if previous_level is not None and level <= previous_level:
+                diff = previous_level - level + 1
+                for i in range(diff):
```

Comment:
Either way is fine by me but maybe this would be simpler?
`name_stack = name_stack[:-diff]`

Reply:
yea, you're right. I'll update it

Reply:
yea, you're right. I'll update it